ui/components/src/ui-components/inputs/QuantityInput.tsx

```diff
@@ -0,0 +1,84 @@
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Inputs
+ */
+
+import * as React from "react";
+import classnames from "classnames";
+import { CommonProps } from "@bentley/ui-core";
+import { ParseResults } from "@bentley/ui-abstract";
+import { QuantityStatus } from "@bentley/imodeljs-quantity";
+import { IModelApp, QuantityType } from "@bentley/imodeljs-frontend";
+import { ParsedInput } from "./ParsedInput";
+import { UiComponents } from "../UiComponents";
+
+/** Props for [[QuantityInput]] control
+ * @beta
+ */
+export interface QuantityProps extends CommonProps {
+  /** Initial magnitude in 'persistence' units. See `getUnitByQuantityType` in [QuantityFormatter]($imodeljs-frontend) */
+  initialValue: number;
+  /** Type of quantity being input. */
+  quantityType: QuantityType;
+  /** Function to call in the quantity value is changed. The value returned will be in 'persistence' units. */
+  onQuantityChange: (newQuantityValue: number) => void;
+  /** Set to `true` if value is for display only */
+  readonly?: boolean;
+}
+
+/** Input control that allows users to input a quantity and show the formatted string that represents the value.
+ * @beta
+ */
+export const QuantityInput = React.forwardRef<HTMLInputElement, QuantityProps>(
+  function QuantityInput({ initialValue, quantityType, readonly, className, style, onQuantityChange }, ref) {
+    const [formatterSpec, setFormatterSpec] = React.useState(() => IModelApp.quantityFormatter.findFormatterSpecByQuantityType(quantityType));
+    const [parserSpec, setParserSpec] = React.useState(() => IModelApp.quantityFormatter.findParserSpecByQuantityType(quantityType));
+
+    const formatValue = React.useCallback((value: number) => {
+      // istanbul ignore else
+      if (formatterSpec) {
+        return IModelApp.quantityFormatter.formatQuantity(value, formatterSpec);
+      }
+      // istanbul ignore next
+      return value.toFixed(2);
+    }, [formatterSpec]);
+
+    const parseString = (userInput: string): ParseResults => {
+      // istanbul ignore else
+      if (parserSpec) {
+        const parseResult = IModelApp.quantityFormatter.parseIntoQuantityValue(userInput, parserSpec);
+        // istanbul ignore else
+        if (parseResult.status === QuantityStatus.Success) {
+          return { value: parseResult.value };
+        } else {
+          return { parseError: `${UiComponents.translate("QuantityInput.NoParserDefined")}${parseResult.status}` };
+        }
+      }
+      // istanbul ignore next
+      return { parseError: UiComponents.translate("QuantityInput.NoParserDefined") };
+    };
+
+    const handleQuantityChange = React.useCallback((newValue: number) => {
+      onQuantityChange && onQuantityChange(newValue);
+    }, [onQuantityChange]);

```

Comment:
Just pass `onQuantityChange` down to `ParsedInput` (onQuantityChange prop is also always defined).
```suggestion
```